### PR TITLE
Style the redirect form

### PR DESCRIPTION
**Pull request by MKodde**
## Background
This is the form EB pages use to send back SAML messages (supports both SAMLResponse and SAMLRequest) but I only know of it using it to send back SAMLResponses to the SP.

## Test notes
I added a functional testing endpoint to render the form. In order to prevent the infinite loop and test other stuff:

1. Manually remove the auto submit code from the body tag of the twig template to prevent a submit loop
2. You can test no-js capabilities by disabling JS in your browser (duh)

See: `https://engine.vm.openconext.org/functional-testing/send`

## Screenshots

Mobile:
Screenshot from 2020-10-01 12-58-22

Larger browser screen:
Screenshot from 2020-10-01 12-50-25

No-JS
Screenshot from 2020-10-01 12-55-05

## Remarks
I'm aware the no-js styling is lacking in theming, and does not have good a11y feature (like outline on focus). For now I'd like to await a first review of the basic styling.

### CODE REVIEW

File: theme/base/stylesheets/pages/form.scss [hunk 1]
```diff
@@ -0,0 +1,22 @@
+/*******************************************************************
+    CSS specific to the form page (auto submitted page that sends
+    SAML response to SP
+ ******************************************************************/
+
+.form {
+    > .form__content {
+        color: $loginBlue;
+        font-size: 16px;
```
Comment (Badlapje) [inline theme/base/stylesheets/pages/form.scss:9]: $f-normal is beter.  Die gebruikt relatieve waarden in plaats van een absolute waarde en is dus beter voor zoom.

File: theme/base/stylesheets/pages/form.scss [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        padding: 0 2rem calculateRem(40px);
```
Comment (Badlapje) [inline theme/base/stylesheets/pages/form.scss:10]: niet zo belangrijk, maar je kan de calculateRem call vervangen door 2.5rem, en die van 20px door 1.25 rem.  Bespaart weer wat rekenkracht.

File: theme/base/stylesheets/pages/form.scss [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        @include screen('mobile') {
+            padding: 0 1rem calculateRem(20px);
+        }
+
+        > picture.form__spinner {
+            > img {
+                padding: 2em 2em 0 0;
```
Comment (Badlapje) [inline theme/base/stylesheets/pages/form.scss:17]: kies je hier bewust voor em?  Anders zou ik rem nemen.  Zie ook [deze link](https://zellwk.com/blog/rem-vs-em/): em leidt heel vaak tot verwarring omdat het in hetzelfde code-blok andere waarden kan aannemen + het kan onbedoeld veranderen als iemand de parent font size aanpast zonder te kijken down-stream.

File: theme/base/templates/modules/Authentication/View/Proxy/form.html.twig [hunk 4]
```diff
@@ -2,39 +2,51 @@
...
-    <html>
+    <html lang="{{ locale()|escape('html_attr') }}">
     <head>
+        <title>{{ defaultTitle }} - {{ 'post_data'|trans }}</title>
         <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"/>
         <meta name="robots" content="noindex, nofollow"/>
-        <meta content="ie=edge,chrome=1" http-equiv="x-ua-compatible">
-        <meta content="initial-scale=1.0,user-scalable=no,maximum-scale=1,width=device-width" name="viewport">
+        <meta content="IE=edge" http-equiv="x-ua-compatible">
+        <meta content="initial-scale=1.0,width=device-width" name="viewport">
```
Comment (Badlapje) [inline theme/base/templates/modules/Authentication/View/Proxy/form.html.twig:11]: 🏅

File: theme/base/templates/modules/Authentication/View/Proxy/form.html.twig [hunk 5]
```diff
@@ -2,39 +2,51 @@
...
+
+    <div class="container form">
 
-        <h1>{{ 'processing'|trans }}</h1>
+        <header>
+            <h1 class="redirectBar">{{ 'processing'|trans }}</h1>
+        </header>
 
-        <p>{{ 'processing_waiting'|trans }}</p>
+        <div class="form__content">
```
Comment (Badlapje) [inline theme/base/templates/modules/Authentication/View/Proxy/form.html.twig:28]: you could change this one out for a main element

File: theme/base/templates/modules/Authentication/View/Proxy/form.html.twig [hunk 6]
```diff
@@ -2,39 +2,51 @@
...
-                {{ 'note_no_script'|trans }}
-            </p>
-        </noscript>
-        <form id="ProcessForm" method="post" action="{{ action }}">
-            <input type="hidden" name="{{ name }}" value="{{ message }}"/>
+            <p>{{ 'processing_waiting'|trans }}</p>
+            <p>{{ 'processing_long'|trans }}</p>
 
-            {{ xtra|raw }}
+            <picture class="form__spinner">
```
Comment (Badlapje) [inline theme/base/templates/modules/Authentication/View/Proxy/form.html.twig:33]: if we don't use another image source, then using a plain img tag is better

File: theme/base/templates/modules/Authentication/View/Proxy/form.html.twig [hunk 7]
```diff
@@ -2,39 +2,51 @@
...
+                <source srcset="" media="(min-width: 800px)"/>
+                <img src="/images/spinner.svg" alt="">
+            </picture>
 
-            <noscript><input type="submit" value="Submit" class="c-button"/></noscript>
-        </form>
+            <noscript>
+                <p>
```
Comment (Badlapje) [inline theme/base/templates/modules/Authentication/View/Proxy/form.html.twig:39]: you could move this to the noscript tag of the form itself, to save one noscript tag & put it all together (i'd have to run the html through a validator but 99% sure that's allowed)

File: src/OpenConext/EngineBlockFunctionalTestingBundle/Resources/config/routing.yml [hunk 8]
```diff
@@ -55,6 +55,16 @@ functional_testing_consent:
     defaults:
         _controller: engineblock.functional_test.controller.consent:consentAction
 
+functional_testing_send_form:
```
Comment (Badlapje) [inline src/OpenConext/EngineBlockFunctionalTestingBundle/Resources/config/routing.yml:58]: 🚀

**Comment by Badlapje** [comment]
Not sure the blue font on blue background has enough contrast, i'll check that in 30

File: theme/base/stylesheets/pages/form.scss (hunk 1 above)
Comment (MKodde) [reply theme/base/stylesheets/pages/form.scss:9]: Thanks, got to get this into my system! I'm used to setting everything to an absolute value everywhere possible ;)

File: theme/base/stylesheets/pages/form.scss (hunk 2 above)
Comment (MKodde) [reply theme/base/stylesheets/pages/form.scss:10]: Thanks, I'm all for having a smaller footprint. If for convenience, maybe we should remove the calculate-rem function/mixin? Or is it usefull in some other way?

File: theme/base/stylesheets/pages/form.scss (hunk 3 above)
Comment (MKodde) [reply theme/base/stylesheets/pages/form.scss:17]: Thanks for the link. I now see why using REM is usefull and more predictable overall. I used to use EM for everything without knowing the difference between the two :blush:

File: theme/base/templates/modules/Authentication/View/Proxy/form.html.twig (hunk 5 above)
Comment (MKodde) [reply theme/base/templates/modules/Authentication/View/Proxy/form.html.twig]: The reason I refrained from doing this was that the parent main element is styled to have a white background with the rounded corners at the bottom. As this is a screen that flashes by for most users. I think it's nicer to have this themed less prominently. 

We should verify this with our PM's don't you think?

File: theme/base/templates/modules/Authentication/View/Proxy/form.html.twig (hunk 6 above)
Comment (MKodde) [reply theme/base/templates/modules/Authentication/View/Proxy/form.html.twig:33]: Ah! Cool with me. Copy pasted this from another template. In the hope I was doing the right thing ;)

File: theme/base/templates/modules/Authentication/View/Proxy/form.html.twig (hunk 7 above)
Comment (MKodde) [reply theme/base/templates/modules/Authentication/View/Proxy/form.html.twig]: :+1:

**Comment by MKodde** [comment]
Thanks for the thorough review. I've addressed all of you feedback points. And in regards to:

1. Yes, changed it to the light grey background color
2. Ack
3. Thanks, have to get those mixins into my system!

See screenshot with where we're at at this point. I suggest we show this to the PM's as well. And align their suggestions in a successive PR

Screenshot_2020-10-05 OpenConext - POST Data

File: theme/base/templates/modules/Authentication/View/Proxy/form.html.twig (hunk 5 above)
Comment (Badlapje) [reply theme/base/templates/modules/Authentication/View/Proxy/form.html.twig]: Agreed.

**Review by Badlapje — APPROVED** [review]

**Comment by Badlapje** [comment]
Ziet er goed uit 💯

**Pull request merged**